File: solutions/r/difference-of-squares/1/difference-of-squares.R
# this is a stub function that takes a natural_number
# and should return the difference-of-squares as described
# in the README.md

square_of_sum <- function(n) {
  sum <- n * (n + 1) / 2
  sum * sum
}

sum_of_squares <- function(n) {
  (n * (n + 1) * (2*n + 1)) / 6
}

difference_of_squares <- function(n) {
  square_of_sum(n) - sum_of_squares(n)
}

File: solutions/lua/difference-of-squares/1/difference-of-squares.lua
local function square_of_sum(n)
  local sum = n * (n + 1) / 2
  return sum * sum
end

local function sum_of_squares(n)
  return (n * (n + 1) * (2 * n + 1)) / 6
end

local function difference_of_squares(n)
  return square_of_sum(n) - sum_of_squares(n)
end

return { square_of_sum = square_of_sum, sum_of_squares = sum_of_squares, difference_of_squares = difference_of_squares }
